drivers/entropy/Kconfig.nrf5

```diff
@@ -16,6 +16,7 @@ menuconfig ENTROPY_NRF5_RNG
 	default y
 	depends on !ENTROPY_NRF_FORCE_ALT
 	depends on DT_HAS_NORDIC_NRF_RNG_ENABLED
+	depends on MULTITHREADING # k_sem_xxx() functions used in the driver
```

Comment:
Are those failing in CI? I believe multithreading support is an implicit assumption for a lot of Zephyr drivers, and generally they don't explicitly declare this dependency. I believe what got added here was for a specific CI test case that gets run in single-threaded mode, so it needed this explicit dependency.

Reply:
> Are those failing in CI?

Not in the tests that run for this PR because as of now the CI is green here, but maybe they can fail on the weekly run which includes more tests? I don´t know, but this looks like a reasonable request to me.

> I believe multithreading support is an implicit assumption for a lot of Zephyr drivers, and generally they don't explicitly declare this dependency.

Yes, but I guess that there are a lot of drivers that make use of semaphores/mutexes/etc, but perhaps not in the initialization phase, so that's why they don't fail in the tests as this entropy driver is doing. Or the ones that did fail, now they might have the `depends on` guard properly set. Just guessing though, I didn't check.

> I believe what got added here was for a specific CI test case that gets run in single-threaded mode, so it needed this explicit dependency.

Yes, correct.

Reply:
> but maybe they can fail on the weekly run which includes more tests? I don´t know, but this looks like a reasonable request to me.

Indeed, my guess is that this failure was seen with a particular configuration exposing a new dependency, but that other configurations with the same dependency might live below the radar until they're caught by another CI run.

Anyway, ok, fair enough: Not directly linked so I won't block for this.